twins/twins.go

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright (c) 2019
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package twins
+
+import (
+	"context"
+	"time"
+)
+
+// Metadata stores arbitrary twin data
+type Metadata map[string]interface{}
+
+// Definition stores entity's attributes
+type Definition struct {
+	Revision   int
+	Created    time.Time
+	Attributes map[string]interface{}
+}
+
+// State stores actual snapshot of entity's values
+type State struct {
+	Definition int
+	Serial     int
+	Created    time.Time
+	Payload    map[string]interface{}
+}
+
+// Twin represents a Mainflux thing digital twin. Each twin is owned by one thing, and
+// is assigned with the unique identifier and (temporary) access key.
+type Twin struct {
+	Owner       string
+	ID          string
+	Key         string
```

Comment:
What is a Key ?

Reply:
It's a unique twin key. Dunno if we need it at all... @nmarcetic ?

Reply:
I don't think so. Twin doesn't need a key, because it's just a representation (e.g Twin can't update itself or do anything actually).

Reply:
Removed Key from Twin.